packages/tpetra/core/src/Tpetra_FEMultiVector_def.hpp

```diff
@@ -59,25 +61,28 @@ FEMultiVector(const Teuchos::RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> >
               const bool zeroOut):
   base_type(importer.is_null()? map:importer->getTargetMap(),numVecs,zeroOut),
   importer_(importer) {
+  const char tfecfFuncName[] = "FEMultiVector::FEMultiVector(): ";
 
-  activeMultiVector_ = Teuchos::rcp(new FEWhichActive(FE_ACTIVE_TARGET));
+  activeMultiVector_ = Teuchos::rcp(new FEWhichActive(FE_ACTIVE_OWNED_PLUS_SHARED));
 
   // Sanity check the importer
   if(!importer_.is_null() && !importer_->getSourceMap()->isSameAs(*map)) {
     throw std::runtime_error("FEMultiVector: 'map' must match 'importer->getSourceMap()' if importer is provided");
   }
 
   if(!importer_.is_null()) {
-    // Check maps to see if we can reuse memory (aka do numSames == domainMap->getNodeNumElements())
-    if(importer_->getNumSameIDs() == importer->getSourceMap()->getNodeNumElements()) {
-      //   1) If so, we then build the inactiveMultiVector_ (w/ source map) using a restricted DualView
-      inactiveMultiVector_ = Teuchos::rcp(new base_type(importer_->getSourceMap(),Kokkos::subview(this->view_,Kokkos::pair<size_t,size_t>(0,map->getNodeNumElements()),Kokkos::ALL)));
-    }
-    else {
-      //   2) If not call a new constructor for the inactive guy (w/ source map)
-      inactiveMultiVector_ = Teuchos::rcp(new base_type(importer_->getSourceMap(),numVecs,zeroOut));
+    // The locallyFitted check is debug mode only since it is more expensive
+    const bool debug = ::Tpetra::Details::Behavior::debug ();
+    if(debug) {
+      TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC( !importer->getTargetMap()->isLocallyFitted(*importer->getSourceMap()),
+                                             std::runtime_error,"importer->getTargetMap() must be locally fitted to importer->getSourceMap()");
+      
     }
+
+    // Memory aliasing is required for FEMultiVector
+    inactiveMultiVector_ = Teuchos::rcp(new base_type(importer_->getSourceMap(),Kokkos::subview(this->view_,Kokkos::pair<size_t,size_t>(0,map->getNodeNumElements()),Kokkos::ALL)));
```

Comment:
What about `origView_`?  Also, FEMultiVector could be blocked by #4639.

Reply:
FEMultiVector does not support origView semantics.  That's not to say it shouldn't support it.  It just doesn't.

Reply:
@csiefer2 That's OK, though we might then want to make the relevant methods that depend on `origView_` (`offsetView*`) throw.